commands/ManufacturerSpecific.go

```diff
@@ -8,15 +8,15 @@ import (
 	"github.com/Sirupsen/logrus"
 )
 
-type CmdManufacturerSpecific struct {
+type ManufacturerSpecificReport struct {
+	*report
 	Manufacturer string `json:"manufacturer"`
 	Type         string `json:"type"`
 	Id           string `json:"id"`
 }
 
-func NewCmdManufacturerSpecific(data []byte) *CmdManufacturerSpecific {
+func NewManufacturerSpecificReport(data []byte) *ManufacturerSpecificReport {
 	if len(data) != 6 {
-
 		logrus.Warn("Failed to decode ManufacturerSpecific: Wrong length")
 		return nil
```

Comment:
All of these constructors should be documented. Especially because their behavior is non-standard: instead of returning an error, they return a sentinel nil value which the caller needs to be careful about.